### PR TITLE
Runtime 44

### PR DESCRIPTION
**Pull request by bbhtt**
Closes https://github.com/flathub/org.gnome.Hamster/issues/6
Closes https://github.com/flathub/org.gnome.Hamster/pull/7

Issue for a better icon https://github.com/projecthamster/hamster/issues/724 This one is upscaled from 48x48, a 128x128 icon either svg or png is mandatory but it's not supplied by upstream.

### CODE REVIEW

**Comment by rabin-io** [comment]
@bbhtt The icon is more broken now on the site, do you know to fix it?

**Comment by bbhtt** [comment]
> @bbhtt The icon is more broken now on the site, do you know to fix it?

Can you trigger a build from here https://buildbot.flathub.org/#/apps/org.gnome.Hamster? Looks like @barthalion triggered one but it failed.

If it passes and is published, the icon will be updated. It might take some time for the icon to update because of CDN caching etc.

**Comment by bbhtt** [comment]
Hmm looks like they switched the program, it was using `appstream-compose` for the old builds but for the new ones it is using `appstreamcli compose`

**Comment by bbhtt** [comment]
I asked about the issue in #flatpak:matrix.org. The change is causing other builds to fail as well.

I did a bunch of tinkering with the upstream app data file: https://github.com/bbhtt/hamster/commit/f5ce15f68d524d960dee0577176203cfdfd6bd3d

The only thing that makes `appstreamcli compose` work is if I change the id tag there to match the Flatpak id.

I copied the directory `build/export/share` to `.flatpak-builder/rofiles/rofiles-xxxxxx/files`, then ran appstreamcli compose locally:

```
appstreamcli compose --prefix=/ --origin=org.gnome.Hamster '--components=org.gnome.Hamster,org.gnome.Hamster.desktop' .flatpak-builder/rofiles/rofiles-FsDHqw/files
Automatically selected '.flatpak-builder/rofiles/rofiles-FsDHqw/files' as data output location.
Only accepting components: org.gnome.Hamster, org.gnome.Hamster.desktop
Processing directory: .flatpak-builder/rofiles/rofiles-FsDHqw/files
Composing metadata...
Success!
```

**Comment by bbhtt** [comment]
Also note that this line isn't needed https://github.com/flathub/org.gnome.Hamster/blob/4ea63b479f996a381759fdca2077abc7b241aaa3/org.gnome.Hamster.yaml#L42. Because upstream waf script is installing that file for us 

https://github.com/bbhtt/hamster/blob/d10ae12cde0370233189574d255c831e3eb267ab/data/wscript#L18

I'm not sure why it was there...

File: org.gnome.Hamster.yaml [hunk 1]
```diff
@@ -40,12 +39,14 @@ modules:
...
     sources:
       - type: git
         url: https://github.com/projecthamster/hamster.git
         commit: 70e2547e7b155bda76ceeeb92f26af5bc177fb0a
       - type: patch
         path: patches/00-fix-missing-tags.patch
-      - type: patch
-        path: patches/01-add-remote-icon.patch
+      - type: file
+        path: org.gnome.Hamster.GUI.png
```
Comment (A6GibKm) [inline org.gnome.Hamster.yaml:50]: just use the svg they bundle, this will look bad

Comment (bbhtt) [reply org.gnome.Hamster.yaml:50]: Hmm I didn't notice the svg icon. I'll change it the new PR

Comment (bbhtt) [reply org.gnome.Hamster.yaml:50]: Done in https://github.com/flathub/org.gnome.Hamster/pull/9/commits/0fc475791c6afd7c219f83b82b6283d64c420888

Comment (A6GibKm) [reply org.gnome.Hamster.yaml:50]: Thanks!

**Pull request merged**